modules/apps/27-interchain-accounts/keeper/keeper.go

```diff
@@ -0,0 +1,167 @@
+package keeper
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/cosmos/ibc-go/v2/modules/apps/27-interchain-accounts/types"
+	host "github.com/cosmos/ibc-go/v2/modules/core/24-host"
+)
+
+// Keeper defines the IBC interchain accounts keeper
+type Keeper struct {
+	storeKey   sdk.StoreKey
+	cdc        codec.BinaryCodec
+	portKeeper types.PortKeeper
+}
+
+// NewKeeper creates a new interchain accounts Keeper instance
+func NewKeeper(
+	cdc codec.BinaryCodec, key sdk.StoreKey,
+	portKeeper types.PortKeeper,
+) Keeper {
+	return Keeper{
+		storeKey:   key,
+		cdc:        cdc,
+		portKeeper: portKeeper,
+	}
+}
+
+// Logger returns the application logger, scoped to the associated module
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", fmt.Sprintf("x/%s-%s", host.ModuleName, types.ModuleName))
+}
+
+// prefixStore returns isolated prefix store for each ICA submodule so they can read/write in separate
+// namespace without being able to read/write to each other's data.
+func (k Keeper) prefixStore(ctx sdk.Context, storePrefix string) sdk.KVStore {
+	return prefix.NewStore(ctx.KVStore(k.storeKey), []byte(storePrefix))
+}
+
+// GetAllPorts returns all ports to which a interchain accounts submodule is bound. Used in ExportGenesis
+func (k Keeper) GetAllPorts(ctx sdk.Context, storePrefix string) []string {
```

Comment:
I didn't change any logic in these functions except the first line. Replacing the ctx.KVStore(k.storeKey) with the prefixStore call